packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -110,15 +124,45 @@ class GameWidget<T extends Game> extends StatefulWidget {
     this.focusNode,
     this.autofocus = true,
     MouseCursor? mouseCursor,
-  }) : super(key: key) {
+  })  : create = null,
+        super(key: key) {
     if (mouseCursor != null) {
-      game.mouseCursor = mouseCursor;
+      game!.mouseCursor = mouseCursor;
     }
     if (initialActiveOverlays != null) {
-      game.overlays.addAll(initialActiveOverlays);
+      game!.overlays.addAll(initialActiveOverlays);
     }
   }
 
+  /// Renders a [Game] in a flutter widget tree.
+  ///
+  /// Unlike the default constructor [GameWidget.new], this creates a
+  /// [GameWidget] that controls the creation of the game instance, r
+  /// removing the necessity of rendering this widget inside a [StatefulWidget].
+  ///
+  /// Ex:
+  /// ```
+  /// ...
+  /// Widget build(BuildContext  context) {
+  ///   return GameWidget.controlled(
+  ///     create: MyGameClass.new,
+  ///   )
+  /// }
+  /// ...
+  /// ```
+  const GameWidget.controlled({
```

Comment:
What does `controlled` mean here?

Reply:
This means that the lifespan of the game instance is directly tied to the lifespan of the game widget element. Once it is removed from the tree, the game instance dies.